src/charm.py

```diff
@@ -115,6 +153,17 @@ def _on_start(self, event: StartEvent) -> None:
 
     def _on_update_status(self, _):
         """Handle the update status event"""
+        if (
```

Comment:
Could you externalise this in a method, this is duplicated between config-changed and update-status

Reply:
Can I do in a follow up PR, since I am already working on [DPE-5268](https://warthogs.atlassian.net/browse/DPE-5268) which involves changes to node port code in src/charm.py